1-perfect-hashing/perfect_hashing.cpp

```diff
@@ -0,0 +1,143 @@
+#include<bits/stdc++.h>
+
+using namespace std;
+
+typedef unsigned int uint;
+
+const int PRIME=41,MAX_TEAM_COUNT=37,MAX_PLAYER_COUNT=55;
+
+struct Team
+{
+    string name;
+    vector<string> players;
+};
+
+int n;//count of teams
+Team teams[MAX_TEAM_COUNT];
+//vector with index i consists of strings with hash=i
+vector<string> hashed_players[MAX_PLAYER_COUNT],hashed_teams[MAX_TEAM_COUNT];
+
+void read()
+{
+    //freopen("input.txt","r",stdin);
+    ifstream in;
+    in.open("input.txt");
```

Comment:
Remove commented out line. 
Better use `ifstream in(INPUT_FILENAME);`

Reply:
Comment deleted. 
I've never seen this trick with `ifstream`. Thanks!